chain/cosmos/chain_node.go

```diff
@@ -861,3 +862,73 @@ func (tn *ChainNode) logger() *zap.Logger {
 		zap.String("test", tn.TestName),
 	)
 }
+
+// RegisterICA will attempt to register an interchain account on the counterparty chain.
+func (tn *ChainNode) RegisterICA(ctx context.Context, address, connectionID string) (string, error) {
+	command := []string{tn.Chain.Config().Bin, "tx", "intertx", "register",
+		"--from", address,
+		"--connection-id", connectionID,
+		"--chain-id", tn.Chain.Config().ChainID,
+		"--home", tn.NodeHome(),
+		"--node", fmt.Sprintf("tcp://%s:26657", tn.Name()),
+		"--keyring-backend", keyring.BackendTest,
+		"-y",
+	}
+
+	exitCode, stdout, stderr, err := tn.NodeJob(ctx, command)
+	if err != nil {
+		return "", dockerutil.HandleNodeJobError(exitCode, stdout, stderr, err)
+	}
+	output := IBCTransferTx{}
+	err = yaml.Unmarshal([]byte(stdout), &output)
+	if err != nil {
+		return "", err
+	}
+	return output.TxHash, nil
+}
+
+// QueryICA will query for an interchain account controlled by the specified address on the counterparty chain.
+func (tn *ChainNode) QueryICA(ctx context.Context, connectionID, address string) (string, error) {
+	command := []string{tn.Chain.Config().Bin, "query", "intertx", "interchainaccounts", connectionID, address,
+		"--chain-id", tn.Chain.Config().ChainID,
+		"--home", tn.NodeHome(),
+		"--node", fmt.Sprintf("tcp://%s:26657", tn.Name())}
+
+	exitCode, stdout, stderr, err := tn.NodeJob(ctx, command)
+	if err != nil {
+		return "", dockerutil.HandleNodeJobError(exitCode, stdout, stderr, err)
+	}
+
+	// at this point stdout should look like this:
+	// interchain_account_address: cosmos1p76n3mnanllea4d3av0v0e42tjj03cae06xq8fwn9at587rqp23qvxsv0j
+	// we split the string at the : and then just grab the address before returning.
+	parts := strings.SplitN(stdout, ":", 2)
```

Comment:
[`strings.Fields`](https://pkg.go.dev/strings#Fields) could be appropriate here too. Is there no `--json` option for output on this command?

Reply:
Hmm looking back there is a flag for producing JSON output, must have glossed over it initially.

I'm going to open another PR today that actually adds a test case built on top of this PR so I may actually make this change in that PR.